### PR TITLE
[feat] Native privy auth

### PR DESCRIPTION
**Pull request by coffeexcoin**
Adds a new `AbstractPrivyProvider` under `@abstract-foundation/agw-react/privy`
This provider can be used to wrap the application instead of `AbstractWalletProvider` if the hosting application is using Privy for their own authentication. 

Also added is a convenience hook `useAbstractPrivyLogin` which allows for quick login/link of AGW account on the host application

---

## PR-Codex overview
This PR focuses on integrating a new `Privy` wallet connector into the `agw-react` package, enhancing the login functionality and provider setup for abstract wallet connections.

### Detailed summary
- Added `id: 'xyz.abs.privy'` to the abstract wallet connector.
- Exported `AbstractPrivyProvider` and `useAbstractPrivyLogin` in `privy.ts`.
- Updated connector lookup in `useLoginWithAbstract` to use `id`.
- Implemented `InjectWagmiConnector` for handling wallet connections.
- Created `AbstractPrivyProvider` for configuring the `Privy` provider.
- Introduced `usePrivyCrossAppProvider` for managing cross-app accounts.
- Updated `package.json` to include `@privy-io/react-auth` dependency.

> The following files were skipped due to too many changes: `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/agw-react/src/privy/usePrivyCrossAppProvider.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        case 'wallet_switchEthereumChain':
+          // TODO: do we need to do anything here?
+          return null;
+        case 'wallet_revokePermissions':
+          // TODO: do we need to do anything here?
+          return null;
+        case 'eth_sendTransaction':
+        case 'eth_signTransaction':
+          // TODO: Implement
+          return randomBytes(32).toString('hex'); // fake tx hash
```
Comment (cygaar) [inline packages/agw-react/src/privy/usePrivyCrossAppProvider.ts:161]: What is the implementation going to be here?

File: packages/agw-react/src/privy/usePrivyCrossAppProvider.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        case 'eth_signTypedData_v4':
+          return await signTypedData(
+            JSON.parse(params[1]) as SignTypedDataParams,
+            { address: params[0] },
+          );
+        case 'eth_sign':
+          throw new Error('eth_sign is unsafe and not supported');
+        case 'personal_sign': {
+          console.log(params);
```
Comment (cygaar) [inline packages/agw-react/src/privy/usePrivyCrossAppProvider.ts:170]: Don't need this if you've verified it works

File: packages/agw-react/src/privy/usePrivyCrossAppProvider.ts (hunk 1 above)
Comment (coffeexcoin) [reply packages/agw-react/src/privy/usePrivyCrossAppProvider.ts:161]: tbd - currently only `sendTransaction` is exposed via cross app connect, not a `signTransaction`

This only applies to transactions from the underlying cross connect EOA, not the AGW

Comment (cygaar) [reply packages/agw-react/src/privy/usePrivyCrossAppProvider.ts:161]: ah gotcha

**Pull request merged**